lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2480,31 +2522,40 @@ static void printInstanceOp(OpAsmPrinter &p, Operation *op,
 // MemoryPortOp Custom attr-dict Directive
 //===----------------------------------------------------------------------===//
 
+void MemoryPortOp::getAsmResultNames(
+    function_ref<void(Value, StringRef)> setNameFn) {
+  StringRef base = name();
+  if (base.empty())
+    base = "memport";
+  setNameFn(data(), base);
+  setNameFn(port(), (base + "_port").str());
+}
+
 static ParseResult parseMemoryPortOp(OpAsmParser &parser,
                                      NamedAttrList &resultAttrs) {
-  return parseElideAnnotations(parser, resultAttrs);
+  return parseImplicitSSAName(parser, resultAttrs);
 }
 
 /// Always elide "direction" and elide "annotations" if it exists or
 /// if it is empty.
 static void printMemoryPortOp(OpAsmPrinter &p, Operation *op,
                               DictionaryAttr attr) {
   // "direction" is always elided.
-  printElideAnnotations(p, op, attr, {"direction"});
+  printImplicitSSAName(p, op, attr, {"direction"});
```

Comment:
I'm thinking that its too weird to try to combine `ImplicitSSAName` with multiple results.  Right now it infers the port name from the first result, and its pretty confusing:
```mlir
%myport, %myport_port = firrtl.memoryport Infer %mem : (!firrtl.cmemory<uint<8>, 8>) -> (!firrtl.uint<8>, !firrtl.cmemoryport)
```
Might make more sense as:
```mlir
%myport_data, %myport_port = firrtl.memoryport Infer %mem {name = "myport"} : (!firrtl.cmemory<uint<8>, 8>) -> (!firrtl.uint<8>, !firrtl.cmemoryport)
```
Any preference?

Reply:
I went with the second one